java-checks/src/main/resources/org/sonar/l10n/java/rules/java/S6377_java.html

```diff
@@ -1,18 +1,42 @@
-<p>XML document can be signed to ensure data integrity and authentication. The signature should be verified and validated to make sure it’s secure.
-For instance, signatures based on weak cipher algorithms like MD5 should be rejected, and an XML document should not contain hostile constructs that
-can lead to Denial of Services attacks, like a large number of <code>SignedInfo</code> elements.</p>
+<p>XML signature validations work by parsing third-party data that cannot be trusted until it is actually validated.</p>
```

Comment:
Rule S6377 has been rewritten to better explain its implementation. It's expected.